### PR TITLE
Fix Timer leak when send link remotely closed.

### PR DESCRIPTION
**Pull request by conniey**
When an AMQP link is idle for more than 10 minutes, it is remotely closed by the service. We did not dispose of the `Timer` created in ReactorSender, causing the thread to live forever.  This PR:
* Removes the send timer and replaces it with the Reactor owned `Schedulers.parallel()`.
  *  The purpose of the timer was to check if the message had been acknowledged from the service after a period of time. We do not need a dedicated thread for it.

Additionally:
* Adds missing CHANGELOG in both azure-messaging-eventhubs and azure-core-amqp.
* Removes unused PREFETCH variable in EventProcessorClient.

### CODE REVIEW

File: sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java [hunk 1]
```diff
@@ -56,6 +58,7 @@
     private final ConcurrentMap<String, LinkSubscription<AmqpSendLink>> openSendLinks = new ConcurrentHashMap<>();
     private final ConcurrentMap<String, LinkSubscription<AmqpReceiveLink>> openReceiveLinks = new ConcurrentHashMap<>();
 
+    private final Scheduler timeoutScheduler = Schedulers.parallel();
```
Comment (alzimmermsft) [inline sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java:61]: Was `Schedulers.boundedElastic` investigated as a potential solution as well?

**Review by alzimmermsft — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

File: sdk/core/azure-core-amqp/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## 2.4.0-beta.1 (Unreleased)
 
+### Bug Fixes
+
+- Fixed a bug where SendTimeout-timer thread was not being disposed of.
```
Comment (joshfree) [inline sdk/core/azure-core-amqp/CHANGELOG.md:7]: nit: can we add 2-3 words on how this bug manifests to the end user

**Review by joshfree — APPROVED** [review]

**Pull request merged**